app_flutter/test/build_dashboard_page_test.dart

```diff
@@ -75,6 +75,35 @@ void main() {
     expect(find.byIcon(Icons.settings), findsOneWidget);
   });
 
+  testWidgets('shows key button & legend', (WidgetTester tester) async {
+    final BuildState fakeBuildState = FakeBuildState();
+
+    await tester.pumpWidget(
+      MaterialApp(
+        home: ValueProvider<BuildState>(
+          value: fakeBuildState,
+          child: ValueProvider<GoogleSignInService>(
+            value: fakeBuildState.authService,
+            child: const BuildDashboardPage(),
+          ),
+        ),
+      ),
+    );
+
+    expect(find.byIcon(Icons.vpn_key), findsOneWidget);
+
+    await tester.tap(find.byIcon(Icons.vpn_key));
+    await tester.pump();
+
+    expect(find.text('Failed'), findsOneWidget);
```

Comment:
You can use table driven design to loop TaskBox.statusColors.keys to validate all are being shown.